tests/python/pants_test/auth/test_basic_auth.py

```diff
@@ -0,0 +1,80 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import base64
+import os
+import threading
+
+from six.moves import BaseHTTPServer
+
+from pants.auth.basic_auth import BasicAuth, BasicAuthCreds
+from pants.auth.cookies import Cookies
+from pants.util.contextutil import environment_as, temporary_dir
+from pants_test.test_base import TestBase
+
+
+class RequestHandlerForTest(BaseHTTPServer.BaseHTTPRequestHandler):
+  def do_GET(self):
+    auth_header = self.headers.get('Authorization')
+    assert auth_header is not None
+    token_type, _, credentials = auth_header.partition(' ')
+    assert token_type == 'Basic'
+    username, password = base64.b64decode(credentials).decode('utf8').split(':')
+    assert username == 'test_user'
+    assert password == 'test_password'
+    self.send_response(200)
+    self.send_header('Set-Cookie', 'test_auth_key=test_auth_value; Max-Age=3600')
+    self.end_headers()
+
+def _run_test_server():
+  httpd = BaseHTTPServer.HTTPServer(('localhost', 0), RequestHandlerForTest)
+  thread = threading.Thread(target=httpd.serve_forever)
+  thread.daemon = True
```

Comment:
This is good, but it would be nice to also return `httpd.shutdown` so that `setUp` could `self.addCleanup(shutdown)` for crisper cleanup of stray threads.

Reply:
Nice idea! Done.